store/tikv/2pc.go

```diff
@@ -359,6 +249,7 @@ func (c *twoPhaseCommitter) initKeysAndMutations() error {
 		for i, op := range mutations.ops {
 			if op != pb.Op_CheckNotExists {
 				c.primaryKey = mutations.keys[i]
+				c.primaryIdx = i
```

Comment:
The `stripNoNeedCommitKeys` may change the `mutations` array, after it's changed the `primaryIdx` may become invalid. Better to use `c.primaryKey`  during comparison to avoid this risk.

Reply:
We need the index in order to strip the primary key out of the array of keys. We only use secondaries in prewrite, which must happen before stripNoNeedCommitKeys, so it seems to me this should be ok?